ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,211 @@
+    output_path = Path(help="Merged-DL1 output filename").tag(config=True)
+    skip_images = Bool(help="Skip DL1/Event/Image data in output",
+                       default_value=False).tag(config=True)
+    skip_parameters = Bool(help="Skip image parameters",
+                           default_value=False).tag(config=True)
+    overwrite = Bool(help="overwrite output file if it exists").tag(config=True)
+    progress_bar = Bool(help="show progress bar during processing").tag(config=True)
+
+    aliases = {'input': 'DL1MergeTool.input_dir',
+               'output': 'DL1MergeTool.output_path'}
+
+    flags = {
+        "skip-images": (
+            {"DL1MergeTool": {"skip_images": True}},
+            "store DL1/Event/Telescope images in output",
+        ),
+        "skip-parameters": (
+            {"DL1MergeTool": {"skip_parameters": True}},
+            "store DL1/Event/Telescope parameters in output",
+        ),
+        "overwrite": (
+            {"DL1MergeTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"DL1MergeTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        )
+    }
+
+    def setup(self):
+        # prepare output path:
+        self.output_path = self.output_path.expanduser()
+        if self.output_path.exists():
+            if self.overwrite:
+                self.log.warning(f"Overwriting {self.output_path}")
+                self.output_path.unlink()
+            else:
+                self.log.critical(f"Output file {self.output_path} exists, "
+                                   "use `--overwrite` to overwrite")
+                sys.exit(1)
+
+        if self.skip_parameters is True and self.skip_images is True:
+            self.log.warning('Skip-parameters and skip-images are both'
+                             'set to True')
+
+        self.input_path_list = sorted(self.input_dir.iterdir())
+
+        # create output file with subarray from first file
+        self.first_subarray = SubarrayDescription.from_hdf(self.input_path_list[0])
+        self.first_subarray.to_hdf(self.output_path)
+        self.output_file = tables.open_file(self.output_path, mode='a')
+
+    def add_image_statistics(self, file):
+        # Creates table for image statistics and adds the entries together.
+        # This does not append rows to the existing table
+        if table_path_image_statistics not in self.output_file:
+            head, tail = os.path.split(os.path.split(table_path_image_statistics)[0])
+            target_group = self.output_file.create_group(head, tail, createparents=True)
+            file.copy_node(table_path_image_statistics, newparent=target_group)
+            file.copy_node('/dl1/service/image_statistics.__table_column_meta__',
+                           newparent=target_group)
+        elif table_path_image_statistics in file:
+            for row in range(3):
```

Comment:
assigning the table to a local variable like this might make it more readable:

```
table  = self.output_file.root[table_path_image_statistics]
```
```